packages/block-library/src/image/image.js

```diff
@@ -745,20 +745,24 @@ export default function Image( {
 							resetLightbox={ resetLightbox }
 						/>
 					) }
-					{ allowCrop && (
-						<ToolbarButton
-							onClick={ () => setIsEditingImage( true ) }
-							icon={ crop }
-							label={ __( 'Crop' ) }
-						/>
-					) }
-					{ showCoverControls && (
-						<ToolbarButton
-							icon={ overlayText }
-							label={ __( 'Add text over image' ) }
-							onClick={ switchToCover }
-						/>
-					) }
+					{ allowCrop &&
+						! isContentOnlyMode &&
+						hasNonContentControls && (
+							<ToolbarButton
+								onClick={ () => setIsEditingImage( true ) }
+								icon={ crop }
+								label={ __( 'Crop' ) }
+							/>
+						) }
+					{ showCoverControls &&
+						! isContentOnlyMode &&
+						hasNonContentControls && (
```

Comment:
Here too, we should update the conditional statement for the `showCoverControls` variable.